### PR TITLE
docs: add Gateway contracts, link to corresponding doc pages

### LINKED ISSUE #363: Add relay and gateway addresses to contracts page
Self-explanatory, should be a simple change we can create a separate section

### PR DESCRIPTION
**Pull request by derrekcoleman**
- Add BOB Gateway contracts to Contracts page
- Hyperlink relay and gateway words to corresponding docs (mainly so people don't think tBTC relay is tBTC token)
- Rearrange sections to put gateway and relay on top, then BOB, then ethereum (same for testnet section). This makes the diff look crazier than it is
- Add missing code format `` around address to the only section that didn't have it

Please take a quick peek at the Vercel preview to sanity check links, formatting 🙏 

Closes #363

### CODE REVIEW

File: docs/docs/build/contracts/index.md [hunk 1]
```diff
@@ -2,27 +2,15 @@
...
-- L1CrossDomainMessenger: [0x237853621998a33Fa5B9B820592F4c6f4c158c12](https://etherscan.io/address/0x237853621998a33Fa5B9B820592F4c6f4c158c12)
-- ProxyAdmin: [0x0d9f416260598313Be6FDf6B010f2FbC34957Cd0](https://etherscan.io/address/0x0d9f416260598313Be6FDf6B010f2FbC34957Cd0)
-- L1ERC721BridgeProxy: [0x5fF93263D5181b2A826f8c51d54BC0da2d20D50a](https://etherscan.io/address/0x5fF93263D5181b2A826f8c51d54BC0da2d20D50a)
-- OptimismPortalProxy: [0x8AdeE124447435fE03e3CD24dF3f4cAE32E65a3E](https://etherscan.io/address/0x8AdeE124447435fE03e3CD24dF3f4cAE32E65a3E)
-- OptimismMintableERC20Factory: [0x0640c9377a4fAC00225097EACbB0295f4D5fbd76](https://etherscan.io/address/0x0640c9377a4fAC00225097EACbB0295f4D5fbd76)
-- L1StandardBridge: [0xEEAfA156A5dd3811Ee0D9F91db57A77eA53A8d31](https://etherscan.io/address/0xEEAfA156A5dd3811Ee0D9F91db57A77eA53A8d31)
-- L1ERC721Bridge: [0xA67fc05D295d2Be6bC4cBfad4f880c1c58734212](https://etherscan.io/address/0xA67fc05D295d2Be6bC4cBfad4f880c1c58734212)
+- [tBTC](/docs/build/bob-sdk/relay) Relay: [`0x9fe7ef727da3d79e0308ff43f31ea1d077ee9f41`](https://explorer.gobob.xyz/address/0x9fe7ef727da3d79e0308ff43f31ea1d077ee9f41)
```
Comment (gregdhill) [inline docs/docs/build/contracts/index.md:7]:
```suggestion
- [Light Relay](/docs/build/bob-sdk/relay) Relay: [`0x9fe7ef727da3d79e0308ff43f31ea1d077ee9f41`](https://explorer.gobob.xyz/address/0x9fe7ef727da3d79e0308ff43f31ea1d077ee9f41)
```

File: docs/docs/build/contracts/index.md [hunk 2]
```diff
@@ -2,27 +2,15 @@
...
+- [Gateway](/docs/build/bob-sdk/gateway): [`0xcc2c1d45A3ECeEdec364a6B9e537AE97Fa20bEa7`](https://explorer.gobob.xyz/address/0xcc2c1d45A3ECeEdec364a6B9e537AE97Fa20bEa7)
```
Comment (gregdhill) [inline docs/docs/build/contracts/index.md:8]: Not sure about linking this, this is only one of many

Comment (derrekcoleman) [reply docs/docs/build/contracts/index.md:8]: Great catch, @gregdhill. I should have checked whether it was an LP-specific address.

Which of [these contracts](https://github.com/bob-collective/bob-gateway#contracts), if any, should we add then?

Comment (derrekcoleman) [reply docs/docs/build/contracts/index.md:8]: @gregdhill, please take a look when you get a chance so we can close this PR 🙏

**Pull request merged**